### PR TITLE
board/Board.mk: Make romfs_img 4 byte aligned

### PR DESCRIPTION
**Pull request by no1wudi**
## Impact
None
## Testing
CI and custom board

### CODE REVIEW

File: boards/Board.mk [hunk 1]
```diff
@@ -36,7 +36,7 @@ $(ETCSRC): $(RCRAWS) $(RCOBJS)
 	  $(shell mkdir -p $(dir $(ETCDIR)$(DELIM)$(raw))) \
 	  $(shell cp -rfp $(raw) $(ETCDIR)$(DELIM)$(raw)))
 	$(Q) genromfs -f romfs.img -d $(ETCDIR)$(DELIM)$(CONFIG_NSH_ROMFSMOUNTPT) -V "$(basename $<)"
-	$(Q) xxd -i romfs.img | sed -e "s/^unsigned/const unsigned/g" > $@
+	$(Q) xxd -i romfs.img | sed -e "s/^unsigned char/const unsigned char __attribute__((aligned(4)))/g" > $@
```
Comment (pkarashchenko) [inline boards/Board.mk:39]: Should we used `aligned_data` from compiler.mk?

Comment (no1wudi) [reply boards/Board.mk:39]: I see aligned_data is from compiler.h, but the etctmp.c is generated dynamically and wouldn't include any header.

Comment (xiaoxiang781216) [reply boards/Board.mk:39]: It's hard or unnecessary to include compiler.h for the dynamic generated code.

Comment (pkarashchenko) [reply boards/Board.mk:39]: I'm just thinking that maybe different alignment is needed for different targets.
Isn't the proper way to set alignment in linker script? Like
```
. = ALIGN (4);
KEEP (*(.romfs_img))
```
???

Comment (pkarashchenko) [reply boards/Board.mk:39]: > It's hard or unnecessary to include compiler.h to use aligned_data for the dynamic generated code.

I think it should be easy to change
```
$(Q) echo "#include <compiler.h>" > $@
$(Q) xxd -i romfs.img | sed -e "s/^unsigned char/const unsigned char aligned_data(4)/g" >> $@
```

Comment (no1wudi) [reply boards/Board.mk]: Thanks, done

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by pkarashchenko** [comment]
@no1wudi shouldn't https://github.com/apache/incubator-nuttx/blob/master/tools/mkromfsimg.sh he updated as well?

**Pull request merged**